### DISCUSSION TITLE
Custom fields dont show

### DISCUSSION THREAD

**Discussion by sophieatgumpo** [woocommerce/woocommerce#53560] [Checkout Flow]
When using the woocommerce checkout widget - the custom fields dont show - local pick up also no longer works with the traditional woocommerce shortcodes - so I have an issue with a client who needs local pick up and custom fields on their checkout.

**Comment by nielslange** [comment]
Thanks for reaching out, @sophieatgumpo. What custom fields do you want to display on the checkout page? You might be able to display them using our [Additional Checkout Fields API](https://developer.woocommerce.com/docs/cart-and-checkout-additional-checkout-fields/).

If that’s not an option, and you need to use the shortcode-based checkout page, consider the classic local pickup option as described in our [Local Pickup Documentation](https://woocommerce.com/document/local-pickup/). Please note that to use the classic local pickup, the checkout page you define in `/wp-admin/admin.php?page=wc-settings&tab=advanced` must use the shortcode-based checkout page.

**Reply by sophieatgumpo** [reply]
Thanks Neil, Local pick up as a shipping method is no longer available when reverting back to the WooCommerce shortcode (see screenshot), the documentation is no longer valid.
![image](https://github.com/user-attachments/assets/73d94edd-cae0-4540-a454-61e9824fd0ba)


We've managed to add the custom field to the Gutenberg checkout using the additional fields documentation - although id like this to only appear on the shipping address not billing as it isn't relevant if the custom has chosen local pick up. I couldn't see how to achieve this in the documentation. Could you assist  me in any way?

**Reply by nielslange** [reply]
> Thanks Neil, Local pick up as a shipping method is no longer available when reverting back to the WooCommerce shortcode (see screenshot), the documentation is no longer valid.

@sophieatgumpo The classic Local Pickup option is only available when you define the page, that contains the checkout shortcode, as the checkout page in ` /wp-admin/admin.php?page=wc-settings&tab=advanced`.

<img width="2559" alt="Screenshot 2024-12-11 at 11 38 21" src="https://github.com/user-attachments/assets/fbc235b1-da13-4033-a088-4b451c5c2860">

> We've managed to add the custom field to the Gutenberg checkout using the additional fields documentation - although id like this to only appear on the shipping address not billing as it isn't relevant if the custom has chosen local pick up. I couldn't see how to achieve this in the documentation. Could you assist me in any way?

What field exactly are you trying to add? Maybe you can share a screenshot. This would help to understand your use case better.

**Reply by sophieatgumpo** [reply]
Thank you for clarification on the classic local pick up - even after I created the old local pickup shipping method it didn't show as an option on the checkout - regardless I'm almost satisfied with the gutenberg checkout.

So it currently showing on both shipping and billing address fields 
![image (5)](https://github.com/user-attachments/assets/c3e4c968-53f2-4510-b380-489c30b1423f)
![image (4)](https://github.com/user-attachments/assets/df13fdc5-777f-4e98-a464-f729dde21f68)

I only want to show this field is they are getting it delivered. (billing - not shipping)
here's my code:

```php
add_action(
    'woocommerce_init',
    function() {
        woocommerce_register_additional_checkout_field(
            array(
                'id'          => 'namespace/how-did-you-hear-about-us',
                'label'       => 'Select Address Type',
                'placeholder' => 'Select Address Type',
                'location'    => 'address',
                'required'    => true,
                'type'        => 'select',
                'options'     => [
                    [
                        'value' => 'commercial',
                        'label' => 'Commercial'
                    ],
                    [
                        'value' => 'residential',
                        'label' => 'Residential'
                    ],
                    [
                        'value' => 'farm',
                        'label' => 'Farm'
                    ],
                    [
                        'value' => 'construction',
                        'label' => 'Construction'
                    ],
                    [
                        'value' => 'school',
                        'label' => 'School'
                    ],
                    [
                        'value' => 'church',
                        'label' => 'Church'
                    ],
                ]
            )
        );
    }
);
```

**Reply by nielslange** [reply]
Thank you for clarifying your use case and sharing the details. I understand the need to display the custom field only within the billing address form to avoid confusion.

Currently, it isn't possible to conditionally render a custom field specifically for the billing or shipping address forms via the provided API. However, as a workaround, you can use CSS to hide the field in the shipping form. Here’s an example:

```css
/* Hide the custom field in the shipping address form */
#shipping .wc-block-components-select-input-namespace-how-did-you-hear-about-us  {
    display: none;
}
```
> [!IMPORTANT]
> While this hides the field visually, its value will still be captured in the order data if the shipping form is filled out. If this behavior affects your workflow, consider modifying your PHP code to conditionally exclude the field's registration for shipping addresses entirely.
